File: src/main/java/ViewUpstreamAction.java
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import org.jetbrains.annotations.NotNull;

class ViewUpstreamAction extends AnAction {
    @Override
    public void actionPerformed(@NotNull AnActionEvent anActionEvent) {
        Utils.runCallGraphFromAction(anActionEvent, CanvasConfig.BuildType.UPSTREAM);
    }

    @Override
    public void update(@NotNull AnActionEvent anActionEvent) {
        Utils.setActionEnabledAndVisibleByContext(anActionEvent);
    }
}

File: src/main/kotlin/callgraph/ViewUpstreamAction.kt
package callgraph

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import oldcallgraph.CanvasConfig
import oldcallgraph.Utils

class ViewUpstreamAction: AnAction() {
    override fun actionPerformed(anActionEvent: AnActionEvent) {
        Utils.runCallGraphFromAction(anActionEvent, CanvasConfig.BuildType.UPSTREAM)
    }

    override fun update(anActionEvent: AnActionEvent) {
        Utils.setActionEnabledAndVisibleByContext(anActionEvent)
    }
}
